crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -195,8 +195,9 @@ impl Format<PyFormatContext<'_>> for MaybeParenthesizeExpression<'_> {
                 f.context().source(),
             );
 
-        let has_comments =
-            comments.has_leading(*expression) || comments.has_trailing_own_line(*expression);
+        let node_comments = comments.leading_dangling_trailing(*expression);
+
+        let has_comments = node_comments.has_leading() || node_comments.has_trailing_own_line();
```

Comment:
Thanks @charliermarsh for adding the methods. It makes it straightforward to replace one API with the other :)